### ISSUE TITLE
If BPFFS is not mounted, cilium cannot populate tunnel endpoint map

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#2924]
Reported on slack: https://cilium.slack.com/archives/C1MATJ5U5/p1519410136000570

Mounting BPFFS is optional according to the cilium k8s install documentation here:

https://github.com/cilium/cilium/blob/master/Documentation/kubernetes/install.rst#mounting-the-bpf-fs-optional

However, users have reported that when running without BPFFS, they may see logs like this after deploying cilium:
```
level=error msg="bpf: Unable to update in tunnel endpoint map" error="Unable to get object /sys/fs/bpf/tc/globals/tunnel_endpoint_map: no such file or directory" ipAddr=10.31.0.0/16
```

This appears to be because the generic map `Update()` opens the map by path via this code:

https://github.com/cilium/cilium/blob/583d87ae3ee038047f76113493a674dbc3e1d69b/pkg/bpf/map.go#L355-L375

~~If BPFFS is unavailable, then the map should not be opened by path.~~

Perhaps BPF mount should be shifted earlier in the startup process so that by the time maps need to be used, BPFFS is available.

**Comment by tgraf** [comment]
Is this an order issue that we attempt to update the map before the agent auto mounts `bpffs`?

**Comment by eloycoto** [comment]
FYI,  I remember to have this problem, and I added a PR at that time: https://github.com/cilium/cilium/pull/1460

I think that the issue is that the container mount the filesystem, but the host doesn't have the BPF mount, so cilium will fail.

**Comment by tgraf** [comment]
I've verified that the BPF filesystem is mounted early enough.

**Comment by tgraf** [comment]
Fixed by #3018
